svtyper/singlesample.py

```diff
@@ -0,0 +1,851 @@
+    parser.add_argument('-q', '--sum_quals', action='store_true', required=False, help='add genotyping quality to existing QUAL (default: overwrite QUAL field)')
+    parser.add_argument('--max_reads', metavar='INT', type=int, default=1000, required=False, help='maximum number of reads to assess at any variant (reduces processing time in high-depth regions, default: 1000)')
+    parser.add_argument('--split_weight', metavar='FLOAT', type=float, required=False, default=1, help='weight for split reads [1]')
+    parser.add_argument('--disc_weight', metavar='FLOAT', type=float, required=False, default=1, help='weight for discordant paired-end reads [1]')
+    parser.add_argument('--debug', action='store_true', help=argparse.SUPPRESS)
+    parser.add_argument('--cores', type=int, metavar='INT', required=False, default=None, help='number of workers to use for parallel processing')
+    parser.add_argument('--batch_size', type=int, metavar='INT', required=False, default=1000, help='number of breakpoints to batch for a parallel processing worker job')
+
+    # parse the arguments
+    args = parser.parse_args()
+
+    # if no input, check if part of pipe and if so, read stdin.
+    if args.input_vcf == None:
+        if not sys.stdin.isatty():
+            args.input_vcf = sys.stdin
+
+    # send back the user input
+    return args
+
+def ensure_valid_alignment_file(afile):
+    if not (afile.endswith('.bam') or afile.endswith('.cram')):
+        die('Error: %s is not a valid alignment file (*.bam or *.cram)\n' % afile)
+
+def open_alignment_file(afile, reference_fasta):
+    fd = None
+    if afile.endswith('.bam'):
+        fd = pysam.AlignmentFile(afile, mode='rb')
+    elif afile.endswith('.cram'):
+        fd = pysam.AlignmentFile(afile, mode='rc', reference_filename=reference_fasta)
+    else:
+        die('Error: %s is not a valid alignment file (*.bam or *.cram)\n' % afile)
+
+    return fd
+
+def setup_sample(bam, lib_info_path, reference_fasta, sampling_number, min_aligned):
+    fd = open_alignment_file(bam, reference_fasta)
+
+    # only consider libraries that constitute at least this fraction of the BAM
+    min_lib_prevalence = 1e-3
+
+    sample = None
+    if (lib_info_path is not None) and os.path.exists(lib_info_path):
+        # use pre-calculated library metrics
+        logit('Reading library metrics from %s...' % lib_info_path)
+        with open(lib_info_path, 'r') as f:
+            lib_info = json.load(f)
+            sample = Sample.from_lib_info(fd, lib_info, min_lib_prevalence)
+    else:
+        # calculate and include library metrics from bam/cram
+        sample = Sample.from_bam(fd, sampling_number, min_lib_prevalence)
+
+    sample.set_exp_seq_depth(min_aligned)
+    sample.set_exp_spanning_depth(min_aligned)
+
+    return sample
+
+def dump_library_metrics(lib_info_path, sample):
+    sample_list = [sample]
+    if (lib_info_path is not None) and (not os.path.exists(lib_info_path)):
```

Comment:
Not sure if refusing to write out a new library metrics file if it already exists is the right behavior. It would surprise me.

Reply:
Or this needed for multiprocessing?